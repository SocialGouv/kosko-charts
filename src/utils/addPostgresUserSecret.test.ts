import environmentMock from "@socialgouv/kosko-charts/environments/index.mock";
import { Deployment } from "kubernetes-models/apps/v1/Deployment";

test("should add a generated Postgres user secret to a deployment", async () => {
  jest.doMock("@socialgouv/kosko-charts/environments", () => environmentMock);
  const { addPostgresUserSecret } = await import("./addPostgresUserSecret");
  const deployment = new Deployment({
    spec: {
      selector: { matchLabels: { component: "app" } },
      template: {
        spec: {
          containers: [
            {
              image: "test:42",
              name: "some-container",
            },
          ],
        },
      },
    },
  });
  addPostgresUserSecret(deployment);
  expect(deployment).toMatchSnapshot();
});

test("should add azure-pg-user to a deployment in preproduction", async () => {
  jest.doMock("@socialgouv/kosko-charts/environments", () => () => ({
    ...environmentMock(),
    isPreProduction: true,
  }));
  const { addPostgresUserSecret } = await import("./addPostgresUserSecret");
  const deployment = new Deployment({
    spec: {
      selector: { matchLabels: { component: "app" } },
      template: {
        spec: {
          containers: [
            {
              image: "test:42",
              name: "some-container",
            },
          ],
        },
      },
    },
  });
  addPostgresUserSecret(deployment);
  expect(deployment).toMatchSnapshot();
});

test("should add azure-pg-user to a deployment in production", async () => {
  jest.doMock("@socialgouv/kosko-charts/environments", () => () => ({
    ...environmentMock(),
    isProduction: true,
  }));
  const { addPostgresUserSecret } = await import("./addPostgresUserSecret");
  const deployment = new Deployment({
    spec: {
      selector: { matchLabels: { component: "app" } },
      template: {
        spec: {
          containers: [
            {
              image: "test:42",
              name: "some-container",
            },
          ],
        },
      },
    },
  });
  addPostgresUserSecret(deployment);
  expect(deployment).toMatchSnapshot();
});
