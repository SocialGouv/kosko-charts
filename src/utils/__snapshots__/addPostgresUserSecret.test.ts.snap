// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a generated Postgres user secret to a deployment 1`] = `
Object {
  "apiVersion": "apps/v1",
  "kind": "Deployment",
  "spec": Object {
    "selector": Object {
      "matchLabels": Object {
        "component": "app",
      },
    },
    "template": Object {
      "spec": Object {
        "containers": Array [
          Object {
            "envFrom": Array [
              Object {
                "secretRef": Object {
                  "name": "azure-pg-user-my-test-branch",
                },
              },
            ],
            "image": "test:42",
            "name": "some-container",
          },
        ],
      },
    },
  },
}
`;

exports[`should add pg-user to a deployment in preproduction 1`] = `
Object {
  "apiVersion": "apps/v1",
  "kind": "Deployment",
  "spec": Object {
    "selector": Object {
      "matchLabels": Object {
        "component": "app",
      },
    },
    "template": Object {
      "spec": Object {
        "containers": Array [
          Object {
            "envFrom": Array [
              Object {
                "secretRef": Object {
                  "name": "azure-pg-user-my-test-branch",
                },
              },
            ],
            "image": "test:42",
            "name": "some-container",
          },
        ],
      },
    },
  },
}
`;

exports[`should add pg-user to a deployment in production 1`] = `
Object {
  "apiVersion": "apps/v1",
  "kind": "Deployment",
  "spec": Object {
    "selector": Object {
      "matchLabels": Object {
        "component": "app",
      },
    },
    "template": Object {
      "spec": Object {
        "containers": Array [
          Object {
            "envFrom": Array [
              Object {
                "secretRef": Object {
                  "name": "azure-pg-user-my-test-branch",
                },
              },
            ],
            "image": "test:42",
            "name": "some-container",
          },
        ],
      },
    },
  },
}
`;
